### ISSUE TITLE
`firebase-query` doesn't listen on the `value` event for initialisation

### ISSUE THREAD

**Issue by 43081j** [firebase/polymerfire#43]
If you have a `firebase-query` pointed at a path which doesn't yet exist, it won't update when the first child is added.

This seems to be because the initial `push` to the collection will fire a `value` event, not a `child_added` event.

We should probably be listening on the `value` event when our initial value is `null`, to set the data correctly.

### Example

```html
<firebase-query path="/foo" data="{{data}}"></firebase-query>
```

* `/foo` does not exist in the database
* push to `/foo` (so the first child is created)
* `data` remains empty

### Expected

- The data property is updated

### Actual

- The data property remains unchanged

**Comment by moynob** [comment]
A related problem:

```
<firebase-query path="/foo" data="{{data}}"></firebase-query>
```

If the path `/foo` requires authentication, the state change from "user not authenticated" to "user authenticated" should update the data but this doesn't happen.

**Comment by skamansam** [comment]
I am getting a similar issue witth the following:
```<dom-module id="firetest-username">
<firebase-query 
        path='/users/[[uid]]/info' 
        id="user_doc" 
        data="{{user_data}}">
    </firebase-query>
    <paper-input label="username" id="username" bind-value="{{user_data.username}}"></paper-input>
</dom-module>
<script>
  Polymer({
    is: 'firetest-username',
    properties: {
      uid: String,
      user_data: {
        type: Object,
        observer: 'userDataChanged'
      },
    },
    userDataChanged: function(old_data, new_data){
      console.log('changed: ',old_data, new_data)
    }
});
</script>
```
When the app first loads, it logs `[], undefined`, then after all items are attached, it logs `[], []`. This works when updating the paper-input, but it won't initiall=y populate with the correct values.

**Comment by mbleigh** [comment]
Any push to a collection should fire the `child_added` event. If it's not doing so, there must be something else afoot. Does anyone have a minimal JSBin or similar repro that can demonstrate this so I can try to debug?

**Comment by 43081j** [comment]
@mbleigh im not sure how best to do a jsbin for this as we'd need a firebase with write access on some nodes (so can't use the same as the tests use?).

it seems like if you have the following:

```javascript
{
    "child": {
        "a": true,
        "b": true
    }
}
```

```html
<firebase-document path="/child" data="{{child}}"></firebase-document>
```

and remove both `a` and `b`, a `child-removed` event doesn't fire as the entire object has gone. similarly it seems if it doesn't exist and you push a value, a `child-added` doesn't fire but `value` does.

**Comment by JGSolutions** [comment]
It seems I am having this issues too but when lazy loading the components. If I add firebase-query in the parent it works

Here is my component:
http://jsbin.com/mazamoqabi/edit?html,css,js,output

I have another component that uses firebase-document and it works!

**Comment by 43081j** [comment]
@mbleigh you can likely close this as the original issue was fixed in #60
